teps/0052-nested-triggers.md

```diff
@@ -0,0 +1,97 @@
+---
+status: proposed
+title: Nested Triggers
+creation-date: '2021-02-10'
+last-updated: '2021-02-10'
+authors:
+- '@jmcshane'
+---
+
+# TEP-0052: Nested Triggers
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+<!-- /toc -->
+
+_NB: This document is intended to be a problem statement with a proposed solution.
+This is a first draft of the feature statement for consumption by the broader
+Tekton community._
+
+## Summary
+
+A Trigger in the EventListener at this time can only be invoked through the top
+level sink process. This means that the interceptor logic for any triggered
+resource must be defined in its entirety for every single Trigger loaded in the
+event listener. There should be a way to execute a common set of interceptors before
+branching the interceptor logic to a number of possible `triggertemplate` and
+`triggerbindings` pairings.
+
+## Motivation
+
+The key motivation in this space is to reduce complexity when a number of shared interceptors
+must be invoked before resource creation takes place. If, as is the case with many repositories,
+multiple tasks need to be invoked from the same Git event, each trigger must process a shared
+set of interceptors to determine the event validitiy before invoking the desired resources.
+
+This also allows for shared data to be passed into multiple triggered `TaskRun` or `PipelineRun`
+objects. As a simple example, if shared volumes are desired for a set of tasks or pipelines, 
+today this would need to be done by passing the same data in the incoming event. 
+When processing is centralized, extensions can be set in the interceptor chain and 
+then used by all the downstream resources in processing.
+
+The result of centralizing this processing and passing common data down to downstream pipelines
+is a simpler experience for Triggers authors. It reduces the YAML burden of copying and maintaining
+the same interceptor chain across multiple triggers, it eases the barrier of entry for writing
+new triggers within the system, and it reduces noise in event listeners logs which is helpful
+for debugging issues.
+
+This solution is in line with multiple issues created by Tekton maintainers. We want to solve
+a broad set of use cases by allowing a complex graph of filters and invocations to be invoked
+by a single triggering event.
+
+* [tektoncd/triggers#820](https://github.com/tektoncd/triggers/issues/820)
+* [tektoncd/triggers#367](https://github.com/tektoncd/triggers/issues/367)
+* [tektoncd/triggers#44](https://github.com/tektoncd/triggers/issues/44)
+
+### Goals
+
+* Add a way for `EventListener` triggers and `Triggers` themselves to specify additional triggers
```

Comment:
I'd reframe the Goals section more in terms of the problem than the solution. So something like:  `Allow some shared logic to be processed before Triggers are called`